.github/workflows/build-test-ci.yml

```diff
@@ -17,24 +17,44 @@ jobs:
       matrix:
         os: [ubuntu-latest]
         cc: [gcc, clang]
+        bb: [meson, muon]
     runs-on: ${{ matrix.os }}
     env:
       CC: ${{ matrix.cc }}
+      BB: ${{ matrix.bb }}
     steps:
     - name: Install dependencies
       run: |
         sudo apt-get update
         sudo apt-get install -y python3-pyelftools python3-pip \
                                 libcap-dev libseccomp-dev ninja-build
-        sudo pip3 install meson
+        case "$BB" in
+          meson)
+            sudo pip3 install meson
+            ;;
+          muon)
+            wget https://muon.build/releases/muon-amd64-linux-static -O muon
+            chmod +x muon
+            sudo install -Dm755 muon /usr/local/bin/muon
+            ;;
+        esac
     - uses: actions/checkout@v2
     - run: |
-        meson -Duse_libcap=enabled \
-              -Duse_seccomp=true \
-              -Dbuild_manpages=disabled \
-              -Dtests=true \
-              -Duse_fuzzing=true \
-              build
+        case "$BB" in
+          meson)
+            cmd=meson
+            ;;
+          muon)
+            cmd="muon setup"
+            ;;
+        esac
+
+        $cmd -Duse_libcap=enabled \
+             -Duse_seccomp=true \
+             -Dbuild_manpages=disabled \
+             -Dtests=true \
+             -Duse_fuzzing=true \
+             build
         ninja -C build
```

Comment:
As an extra thought: if we want muon to work, we should probably test with samurai too (small ninja impl)